File: OOP_Variable/3.rb
class Cal
  def initialize(v1, v2)
    @v1 = v1
    @v2 = v2
  end
  def add()
    return @v1 + @v2
  end
  def subtract()
    return @v1 - @v2
  end
  def getVal1()
    return @v1
  end
  def getVal2(v2) #로컬 변수 넣으려면, 사용할때도 받아야 함
    return @v2
  end
  def setVal1(v1)
    if v1.is_a?(Integer)
      @v1 = v1
    end
  end
  def setVal2(v2)
    @v2 = v2
  end
end

c1 = Cal.new(10,20)
p c1.add
p c1.subtract

p c1.setVal1(30)
p c1.add

p c1.getVal1()
# p c1.getVal1(2000000)
p c1.getVal2(20000000)

File: OOP_Variable/3.py
class Cal:
    def __init__(self,v1,v2):
        if isinstance(v1, int):
            self.val1 = v1
        if isinstance(v2, int):
            self.val2 = v2
    def add(v1,v2):
        return v1 + v2
    def add2(self):
        return self.val1 + self.val2
    def subtract(v1, v2):
        return v1 - v2
    def subtract2(self):
        return self.val1 - self.val2
    def setVal1(self,v):
        if isinstance(v, int):
            self.val1 = v
    def getVal1(self):
        return self.val1
    def getVal2(self):
        return self.val2


print(Cal.add(30,20))

# print(Cal.add2(30,20)) # 이건 동작 안함
c1 = Cal(30,20)
print(c1.add2())


# c1.val1 = 'one'
c1.setVal1('one')
c1.val2 = 30

print(c1.add2())



print(Cal.subtract(540, 20))

c2 = Cal(540, 30)
print(c2.subtract2())



print(c1.getVal1())
print(c1.getVal2())
